src/component/registry/types.ts

```diff
@@ -370,8 +371,16 @@ export interface EdgeBpmnSemantic extends BaseBpmnSemantic {
  * @category Custom Behavior
  */
 export interface ShapeBpmnSemantic extends BaseBpmnSemantic {
```

Comment:
**suggestion:** 
Why not have a similar architecture to the internal model, to avoid having to access properties when you know they'll never be fulfilled?

Reply:
I am not sure to understand the remark.
The APIs return a value object. We can introduce various interface (for call activity, event, subprocess) but in the end, this will increase the complexity of the user TypeScript code.
They would have to deal with several types and do some casting to be able to check if the property are set.

Let's talk about this in person.

Reply:
As seen with @csouchet, we find an alternative solution that it is implemented in bf4ac78.
To be reviewed.

Reply:
@tbouffard :pray: for the new implementation